streams/src/main/java/org/apache/kafka/streams/processor/ProcessorContext.java

```diff
@@ -83,19 +83,60 @@
 
     StateStore getStateStore(String name);
 
+    /**
+     * Schedules a periodic operation for processors. A processor may call this method during
+     * {@link Processor#init(ProcessorContext) initialization} to
+     * schedule a periodic call called a punctuation to {@link Processor#punctuate(long)}.
+     *
+     * @param interval the time interval between punctuations
+     */
     void schedule(long interval);
 
+    /**
+     * Forwards a key/value pair to the downstream processors
+     * @param key key
+     * @param value value
+     */
     <K, V> void forward(K key, V value);
 
+    /**
+     * Forwards a key/value pair to one of the downstream processors designated by childIndex
+     * @param key key
+     * @param value value
+     */
     <K, V> void forward(K key, V value, int childIndex);
 
+    /**
+     * Requests a commit
```

Comment:
This is off-topic of this PR: the committing guarantees are a bit in the flux right now: in fact today from a user's point of view calling commit() does not guarantee you anything, since the producer.flush does not wait for acks and hence you could still have data loss. Right?